hathor/transaction/base_transaction.py

```diff
@@ -154,6 +169,10 @@ class BaseTransaction(ABC):
 
     _metadata: Optional[TransactionMetadata]
 
+    # The first byte extracted from the version field. Carries information about Feature Activation
+    # bits and also extra bits reserved for future use, depending on the configuration.
+    signal_bits: int
```

Comment:
Should we add it to `BaseTransaction` or just to `Block`? I mean, does it make sense for transactions to signal supported features?

Reply:
I added it to `BaseTransaction` because it makes sense to extract the `version` field's first byte as a separate field in all cases. I also named it `signal_bits` so it's generic enough, instead of something like `feature_activation_bits`.

It's just an "extra" byte that may be interpreted differently by each subclass of `BaseTransaction`. `Transaction`s may not use them now, but they may in the future (not for feature activation, but for something else).

I've changed the docstrings to make this clearer: https://github.com/HathorNetwork/hathor-core/pull/623/commits/92faeaa168c471e39e7cac465e0b38ec1f4a1a1d

Reply:
If we're changing `TxVersion` we'd need at least to make the implementation of `Transaction` ignore the first byte somehow. I think it's OK to just do it in `BaseTransaction`, since `signal_bits` should be completely ignored for transactions for now, it shouldn't matter much, we can change it later (we could even end up needing it for the next part of Feature Activation design).